Reproducible Failure in org.elasticsearch.cluster.metadata.DateMathExpressionResolverTests.testExpression_MixedArray

This fails and reproduces for me locally 100% of the time on master:

```
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.cluster.metadata.DateMathExpressionResolverTests.testExpression_MixedArray" -Dtests.seed=DD523116F9B24D8C -Dtests.security.manager=true -Dtests.locale=is-IS -Dtests.timezone=Europe/Warsaw -Druntime.java=15

org.elasticsearch.cluster.metadata.DateMathExpressionResolverTests > testExpression_MixedArray FAILED
    java.lang.AssertionError: 
    Expected: ".logstash-2021.01"
         but: was ".logstash-2020.01"
        at __randomizedtesting.SeedInfo.seed([DD523116F9B24D8C:794B3FD6F49CA76C]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.cluster.metadata.DateMathExpressionResolverTests.testExpression_MixedArray(DateMathExpressionResolverTests.java:125)


Suite: Test class org.elasticsearch.cluster.metadata.DateMathExpressionResolverTests
  1> [2021-01-02T16:24:50,530][INFO ][o.e.c.m.DateMathExpressionResolverTests] [testExpression_MixedArray] before test
  1> [2021-01-02T16:24:50,617][INFO ][o.e.c.m.DateMathExpressionResolverTests] [testExpression_MixedArray] after test
  2> REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.cluster.metadata.DateMathExpressionResolverTests.testExpression_MixedArray" -Dtests.seed=DD523116F9B24D8C -Dtests.security.manager=true -Dtests.locale=is-IS -Dtests.timezone=Europe/Warsaw -Druntime.java=15
  2> java.lang.AssertionError: 
    Expected: ".logstash-2021.01"
         but: was ".logstash-2020.01"
        at __randomizedtesting.SeedInfo.seed([DD523116F9B24D8C:794B3FD6F49CA76C]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.cluster.metadata.DateMathExpressionResolverTests.testExpression_MixedArray(DateMathExpressionResolverTests.java:125)
  2> NOTE: leaving temporary files on disk at: /home/brownbear/src/elasticsearch/server/build/testrun/test/temp/org.elasticsearch.cluster.metadata.DateMathExpressionResolverTests_DD523116F9B24D8C-002
  2> NOTE: test params are: codec=Asserting(Lucene87): {}, docValues:{}, maxPointsInLeafNode=878, maxMBSortInHeap=6.27231052818936, sim=Asserting(RandomSimilarity(queryNorm=true): {}), locale=is-IS, timezone=Europe/Warsaw
  2> NOTE: Linux 5.8.0-33-generic amd64/AdoptOpenJDK 15.0.1 (64-bit)/cpus=48,threads=1,free=524039296,total=536870912
  2> NOTE: All tests run in this JVM: [DateMathExpressionResolverTests]

Tests with failures:
 - org.elasticsearch.cluster.metadata.DateMathExpressionResolverTests.testExpression_MixedArray

```

--- a/server/src/test/java/org/elasticsearch/cluster/metadata/DateMathExpressionResolverTests.java
+++ b/server/src/test/java/org/elasticsearch/cluster/metadata/DateMathExpressionResolverTests.java
@@ -120,10 +120,10 @@ public class DateMathExpressionResolverTests extends ESTestCase {
         assertThat(result.size(), equalTo(4));
         assertThat(result.get(0), equalTo("name1"));
         assertThat(result.get(1),
-            equalTo(".marvel-" + DateTimeFormat.forPattern("YYYY.MM.dd").print(new DateTime(context.getStartTime(), UTC))));
+            equalTo(".marvel-" + DateTimeFormat.forPattern("xxxx.MM.dd").print(new DateTime(context.getStartTime(), UTC))));
         assertThat(result.get(2), equalTo("name2"));
         assertThat(result.get(3), equalTo(".logstash-" +
-            DateTimeFormat.forPattern("YYYY.MM").print(new DateTime(context.getStartTime(), UTC).withDayOfMonth(1))));
+            DateTimeFormat.forPattern("xxxx.MM").print(new DateTime(context.getStartTime(), UTC).withDayOfMonth(1))));
     }
 
     public void testExpression_CustomTimeZoneInIndexName() throws Exception {

--- a/server/src/test/java/org/elasticsearch/cluster/metadata/DateMathExpressionResolverTests.java
+++ b/server/src/test/java/org/elasticsearch/cluster/metadata/DateMathExpressionResolverTests.java
@@ -113,18 +113,17 @@ public class DateMathExpressionResolverTests extends ESTestCase {
             equalTo(".marvel-" + DateTimeFormat.forPattern("'{year}'yyyy").print(new DateTime(context.getStartTime(), UTC))));
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/pull/66914")
     public void testExpression_MixedArray() throws Exception {
         List<String> result = expressionResolver.resolve(context, Arrays.asList(
-                "name1", "<.marvel-{now/d}>", "name2", "<.logstash-{now/M{YYYY.MM}}>"
+                "name1", "<.marvel-{now/d}>", "name2", "<.logstash-{now/M{uuuu.MM}}>"
         ));
         assertThat(result.size(), equalTo(4));
         assertThat(result.get(0), equalTo("name1"));
         assertThat(result.get(1),
-            equalTo(".marvel-" + DateTimeFormat.forPattern("xxxx.MM.dd").print(new DateTime(context.getStartTime(), UTC))));
+            equalTo(".marvel-" + DateTimeFormat.forPattern("yyyy.MM.dd").print(new DateTime(context.getStartTime(), UTC))));
         assertThat(result.get(2), equalTo("name2"));
         assertThat(result.get(3), equalTo(".logstash-" +
-            DateTimeFormat.forPattern("xxxx.MM").print(new DateTime(context.getStartTime(), UTC).withDayOfMonth(1))));
+            DateTimeFormat.forPattern("yyyy.MM").print(new DateTime(context.getStartTime(), UTC).withDayOfMonth(1))));
     }
 
     public void testExpression_CustomTimeZoneInIndexName() throws Exception {

